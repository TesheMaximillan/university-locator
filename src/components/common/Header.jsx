/* eslint-disable react/prop-types */
import React from 'react';
import { FaUniversity } from 'react-icons/fa';
import styles from './Header.module.scss';

function Header({ name, view }) {
  const {
    header, image, headerTitle, icon,
  } = styles;

  return (
    <div className={header}>
      <div className={image}>
        <FaUniversity className={icon} />
      </div>
      <div className={headerTitle}>
        <h2>{name}</h2>
        <p>{view}</p>
      </div>
    </div>
  );
}

export default Header;
