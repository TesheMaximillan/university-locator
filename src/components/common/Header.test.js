import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';
import renderContext from '../../redux/storeMock';

describe('Header component', () => {
  let component;

  beforeEach(() => {
    component = renderer.create(renderContext(<Header />));
  });

  it('should render with given state from redux store', () => {
    // Assert
    expect(component.toJSON()).toMatchSnapshot();
  });
});
