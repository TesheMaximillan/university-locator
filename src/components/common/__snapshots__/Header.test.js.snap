// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render with given state from redux store 1`] = `
<div
  className="header"
>
  <div
    className="image"
  >
    <svg
      className="icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M496 128v16a8 8 0 0 1-8 8h-24v12c0 6.627-5.373 12-12 12H60c-6.627 0-12-5.373-12-12v-12H24a8 8 0 0 1-8-8v-16a8 8 0 0 1 4.941-7.392l232-88a7.996 7.996 0 0 1 6.118 0l232 88A8 8 0 0 1 496 128zm-24 304H40c-13.255 0-24 10.745-24 24v16a8 8 0 0 0 8 8h464a8 8 0 0 0 8-8v-16c0-13.255-10.745-24-24-24zM96 192v192H60c-6.627 0-12 5.373-12 12v20h416v-20c0-6.627-5.373-12-12-12h-36V192h-64v192h-64V192h-64v192h-64V192H96z"
      />
    </svg>
  </div>
  <div
    className="headerTitle"
  >
    <h2 />
    <p />
  </div>
</div>
`;
